feature(решение RadioComponentsCollection): добавлен проект Model.UnitTests

TemZZZ opened the pull request:
Юнит-тестирование классов бизнес-логики.

Commit:
fetaure(проект RadioComponentsCollection): добавлен существующий проект UnitTests/UnitTests.csproj

Commit:
Revert "fetaure(проект RadioComponentsCollection): добавлен существующий проект UnitTests/UnitTests.csproj"

This reverts commit 0ee36d1ade5caadcd23bd8a80bce34367cdef7d3.

Commit:
style(решение RadioComponentsCollection): удалены папки UnitTests и packages из папки с решением

Commit:
feature(решение RadioComponentsCollection): добавлен проект UnitTests

- в проект UnitTests добавлена ссылка на проект Model
- в папке проекта UnitTests создана папка Model
- установлен пакет NUnit в проект UnitTests
- установлен пакет NUnit3TestAdapter в проект UnitTests

Commit:
refactor(Testing/Model.UnitTests): переименован из UnitTests

- проект UnitTests переименован в Model.UnitTests
- в решении создана папка Testing
- проект Model.UnitTests перемещен в папку Testing
- из проекта Model.UnitTests удалена папка Model

Commit:
Revert "refactor(Testing/Model.UnitTests): переименован из UnitTests"

This reverts commit 5cd00174ad865d3690aacb760566b3d600e129aa.

Commit:
feature(решение RadioComponentsCollection): удален проект UnitTests

- удалена папка UnitTests со всем содержимым из папки решения
- удалена папка packages со всем содержимым из папки решения

Commit:
feature(решение RadioComponentsCollection): добавлен проект Testing/Model.UnitTests для тестирования классов проета Model

- в папке решения создана папка Testing
- в папке Testing создан проект Model.UnitTests
- в проект Testing/Model.UnitTests добавлена ссылка на проект Model
- установлены пакеты NUnit 3.12 и NUnit3TestAdapter для проекта Testing/Model.UnitTests

Commit:
feature(Model.UnitTests): добавлен класс RadioComponentBaseTests

- в классе RadioComponentBaseTests реализован метод ValueProperty_AssignedGoodValues_IsAssigned

Commit:
test(Model.UnitTests.RadioComponentBaseTests): добавлены тесты положительного сценария для свойства Value

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): реализован тестовый метод ValueProperty_AssignedBadValues_ThrowsExpectedException

Commit:
test(Model.UnitTests.RadioComponentBaseTests): добавлены негативные тесты для свойства Value

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): реализован метод GetImpedance_BadFrequencies_ThrowsExpectedException для тестирования негативных сценариев метода GetImpedance

- в проект Model.UnitTests добавлена ссылка на System.Numerics для возможности возврата типа Complex методом GetImpedance

Commit:
test(Model.UnitTests.RadioComponentBaseTests): добавлены негативные тесты метода GetImpedance

- добавлена константа protected const double MinFrequency = 0

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): добавлен абстрактный метод GetImpedance_GoodValuesAndFrequencies_ReturnsValue

Commit:
style(проект Model.UnitTests, RadioComponentBaseTests.cs): удалены неиспользуемые директивы using

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): добавлен модификатор доступа protected константе MinRadioComponentValue

Commit:
feature(Model.UnitTests): добавлен класс ResistorTests

- реализован метод GetImpedance_GoodValuesAndFrequencies_ReturnsValue

Commit:
test(Model.UnitTests.ResistorTests): добавлены тесты для тестирования метода GetImpedance

iKalentii commented on Testing/Model.UnitTests/RadioComponentBaseTests.cs:
```diff
@@ -0,0 +1,153 @@
+		protected const double MinFrequency = 0;
+
+		/// <summary>
+		/// Когда свойству <see cref="RadioComponentBase.Value"/>
```
XML комментарии в тестах не нужны, т.к. эти методы не вызываются снаружи. Лучше сосредоточиться на названии тестовых случаев.

iKalentii commented on Testing/Model.UnitTests/ResistorTests.cs:
```diff
@@ -0,0 +1,72 @@
+	{
+		static object[] TestCases =
+		{
+			new object[] { MinRadioComponentValue, MinFrequency,
```
Тут ещё не хватает человекопонятного описания тестовых случаев. Что-то такое должно получиться https://stackoverflow.com/questions/26374265/access-nunit-test-name-within-testcasesource

TemZZZ replied:
Да, тоже нашел подобную статейку вот тут: https://software-testing.ru/library/testing/testing-automation/3303-data-driven-testing-in-c-with-nunit-and-restsharp

Commit:
test(Model.UnitTests.ResistorTests): изменены тесты метода GetImpedance

- все тесты теперь формируются методом TestCasesForGetImpedanceMethod
- набор тестов методу GetImpedance_GoodValuesAndFrequencies_ReturnsValue передается через атрибут [TestCaseSource("TestCasesForGetImpedanceMethod")]

Commit:
style(Model.UnitTests.RadioComponentBaseTests): удалены XML-комментарии для методов-тестов

Commit:
feature(Model.RadioComponentBase.Value): добавлена проверка присвоения значения NaN

- при попытке присвоить Value значение NaN выбрасывается исключение ArgumentException
- дополнен XML-комментарий к свойству Value

Commit:
fix(Model.RadioComponentBase.Value): исправлена неправильная проверка значения на NaN

- if (value == double.NaN) исправлена на if (double.IsNaN(value))

Commit:
feature(Model.RadioComponentBase.Value): изменен текст исключения при попытке присвоения отрицательного значения

- "Value must not be less than zero" изменен на "Value of radiocomponent can't be less than zero."

Commit:
feature(Model.RadioComponentBase.GetImpedance): изменен текст исключения при попытке передачи отрицательного значения

Commit:
refactor(проект Model, RadioComponentBase.cs): freq переименован в frequency

Commit:
feature(Model.RadioComponentBase.GetImpedance): добавлена проверка частоты на NaN

- при попытке передачи частоты, равной NaN, выбрасывается исключение ArgumentException
- дополнен комментарий к методу GetImpedance

Commit:
feature(Model.RadioComponentBase.GetImpedance): добавлена проверка на передачу положительной бесконечно большой частоты

- выбрасывается исключение ArgumentOutOfRangeException
- дополнен комментарий к методу

Commit:
feature(Model.RadioComponentBase.Value): добавлена проверка на присвоение положительного бесконечно большого значения

- выбрасывается исключение ArgumentOutOfRangeException
- дополнен комментарий к свойству

Commit:
style(Model.UnitTests.GetImpedance_GoodValuesAndFrequencies_ReturnsValue): удален XML-комментарий

Commit:
style(Model.UnitTests.RadioComponentBaseTests): удален атрибут [Test] у абстрактного метода GetImpedance_GoodValuesAndFrequencies_ReturnsValue

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): добавлены поля-массивы GoodRadioComponentValues и GoodFrequencies

- массив GoodRadioComponentValues содержит набор возможных значений свойства Value
- массив GoodFrequencies содержит набор возможных значений частоты

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): добавлен статический метод ValuePropertyGoodValuesTestCases

- ValuePropertyGoodValuesTestCases создает позитивные тесты для свойства Value
- свойство GoodRadioComponentValues помечено как static

Commit:
refactor(Model.UnitTests.RadioComponentBaseTests): изменены свойства GoodRadioComponentValues и GoodFrequencies

- добавлено приватное статическое поле _goodDoubles
- GoodRadioComponentValues теперь возвращает _goodDoubles
- GoodFrequencies теперь возвращает _goodDoubles

Commit:
test(Model.UnitTests.RadioComponentBaseTests): изменены тесты для ValueProperty_AssignedGoodValues_IsAssigned

- теперь тесты передаются через атрибут TestCaseSource("ValuePropertyGoodValuesTestCases")

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): добавлен приватный словарь _badDoubleToExpectedExceptionTypeMap

- словарь _badDoubleToExpectedExceptionTypeMap ставит в соответствие негативному double требуемый тип исключения

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): реализован метод ValuePropertyBadValuesTestCases

- метод ValuePropertyBadValuesTestCases создает негативные тесты для свойства Value

Commit:
test(Model.UnitTests.RadioComponentBaseTests): удалены негативные тесты для свойства Value

- удалены атрибуты TestCase перед методом ValueProperty_AssignedBadValues_ThrowsExpectedException

Commit:
feature(Model.UnitTests.RadioComponentBaseTests.ValueProperty_AssignedBadValues_ThrowsExpectedException): изменен метод

- теперь метод ValueProperty_AssignedBadValues_ThrowsExpectedException принимает один аргумент KeyValuePair<double, Type> вместо двух double и Type
- тип исключения берется из doubleToExpectedExceptionType.Value
- значение свойства Value берется из doubleToExpectedExceptionType.Key

Commit:
test(Model.UnitTests.RadioComponentBaseTests): добавлены негативные тесты свойства Value

- перед методом ValueProperty_AssignedBadValues_ThrowsExpectedException добавлен атрибут TestCaseSource("ValuePropertyBadValuesTestCases")

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): реализован метод GetImpedanceMethodBadFrequenciesTestCases

- метод GetImpedanceMethodBadFrequenciesTestCases создает негативные тесты для метода GetImpedance

test(Model.UnitTests.RadioComponentBaseTests): удалены негативные тесты для метода GetImpedance

- удалены атрибуты TestCase перед методом GetImpedance_BadFrequencies_ThrowsExpectedException

feature(Model.UnitTests.RadioComponentBaseTests.GetImpedance_BadFrequencies_ThrowsExpectedException): изменен метод

- теперь метод GetImpedance_BadFrequencies_ThrowsExpectedException принимает один аргумент KeyValuePair<double, Type> вместо двух double и Type
- тип исключения берется из doubleToExpectedExceptionType.Value
- значение частоты берется из doubleToExpectedExceptionType.Key

test(Model.UnitTests.RadioComponentBaseTests): добавлены негативные тесты метода GetImpedance

- перед методом GetImpedance_BadFrequencies_ThrowsExpectedException добавлен атрибут TestCaseSource("GetImpedanceMethodBadFrequenciesTestCases")

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): добавлен и реализован метод GetRadioComponent

Commit:
feature(Model.UnitTests.RadioComponentBaseTests): изменены ограничения типа T

- вместо RadioComponentBase и new() теперь IRadioComponent
- изменен комментарий к классу

Commit:
style(Model.UnitTests.RadioComponentBaseTests): изменен комментарий к классу

Commit:
refactor(Model.UnitTests.RadioComponentBaseTests): изменен метод ValueProperty_AssignedGoodValues_IsAssigned

- создание объекта радиокомпонента теперь осуществляется через GetRadioComponent()
- отдельной строкой осуществляется воздействие на объект: radioComponent.Value = value

refactor(Model.UnitTests.RadioComponentBaseTests): изменен метод ValueProperty_AssignedBadValues_ThrowsExpectedException

- создание объекта радиокомпонента теперь осуществляется через GetRadioComponent()
- ожидаемое исключение оформлено отдельной переменной expectedException
- тестовый делегат оформлен отдельной переменной SetRadioComponentValue

refactor(Model.UnitTests.RadioComponentBaseTests): изменен метод GetImpedance_BadFrequencies_ThrowsExpectedException

- создание объекта радиокомпонента теперь осуществляется через GetRadioComponent()
- ожидаемое исключение оформлено отдельной переменной expectedException
- тестовый делегат оформлен отдельной переменной GetRadioComponentImpedance

Commit:
style(проект Model.UnitTests, RadioComponentBaseTests.cs): удалены неиспользуемые директивы using

Commit:
feature(Model.UnitTests.RadioComponentBaseTests.ValuePropertyBadValuesTestCases): изменен аргумент метода SetName

- вместо значения doubleToExpectedExceptionType.Value к имени тестового случая добавляется doubleToExpectedExceptionType.Value.Name

feature(Model.UnitTests.RadioComponentBaseTests.GetImpedanceMethodBadFrequenciesTestCases): изменен аргумент метода SetName

- вместо значения doubleToExpectedExceptionType.Value к имени тестового случая добавляется doubleToExpectedExceptionType.Value.Name

Commit:
refactor(Model.UnitTests.RadioComponentBaseTests.GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance): переименован из GetImpedance_GoodValuesAndFrequencies_ReturnsValue

Commit:
refactor(Model.UnitTests.RadioComponentBaseTests.ValueProperty_AssignedGoodValue_IsAssigned): переименован из ValueProperty_AssignedGoodValues_IsAssigned

Commit:
refactor(Model.UnitTests.RadioComponentBaseTests.ValueProperty_AssignedBadValue_ThrowsExpectedException): переименован из ValueProperty_AssignedBadValues_ThrowsExpectedException

Commit:
refactor(Model.UnitTests.RadioComponentBaseTests.GetImpedance_BadFrequency_ThrowsExpectedException): переименован из GetImpedance_BadFrequencies_ThrowsExpectedException

Commit:
feature(Model.UnitTests.RadioComponentBaseTests.GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance): изменен порядок следования параметров и имена параметров

- value переименован в radioComponentValue
- теперь первым параметром идет frequency, вторым - radioComponentValue

Commit:
refactor(Model.UnitTests.ResistorTests.GetImpedanceMethodTestCases): переименован из TestCasesForGetImpedanceMethod

Commit:
style(Model.UnitTests.ResistorTests.GetImpedanceMethodTestCases): удален XML-комментарий

Commit:
feature(Model.UnitTests.ResistorTests.GetImpedanceMethodTestCases): изменена логика метода

- используются коллекции GoodRadioComponentValues и GoodFrequencies из базового класса
- при создании TestCase-объекта сначала передается frequency, затем radioComponentValue

Commit:
feature(Model.UnitTests.ResistorTests.GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance): изменена реализация метода

- в метод передается сначала frequency, потом radioComponentValue
- объект resistor создается методом GetRadioComponent базового класса
- переменная actual переименована в actualImpedance

Commit:
test(Model.UnitTests.ResistorTests.GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance): изменен TestCaseSource-атрибут

- TestCaseSource("TestCasesForGetImpedanceMethod") изменен на TestCaseSource("GetImpedanceMethodTestCases")

Commit:
fefactor(Model.UnitTests.IRadioComponentTestsBase): переименован из RadioComponentBaseTests

Commit:
style(Model.UnitTests.ResistorTests): класс помечен атрибутом TestFixture

Commit:
style(Model.UnitTests.ResistorTests.GetImpedanceMethodTestCases): добавлен модификатор доступа private

Commit:
feature(Model.UnitTests): добавлен тестовый класс InductorTests

- реализован метод GetImpedanceMethodTestCases, формирующий тесты метода GetImpedance
- реализован тестовый метод GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance
- добавлены тесты метода GetImpedance: перед методом GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance добавлен атрибут [Test, TestCaseSource("GetImpedanceMethodTestCases")]

Commit:
feature(Model.UnitTests.IRadioComponentTestsBase): изменен метод GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance

- метод помечен как виртуальный вместо абстрактного
- добавлена реализация метода

Commit:
refactor(Model.UnitTests.InductorTests.GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance): реализация метода заменена вызовом одноименного метода базового класса

Commit:
refactor(Model.UnitTests.ResistorTests.GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance): реализация метода заменена вызовом одноименного метода базового класса

Commit:
fix(Model.Inductor.CalcImpedance): при частоте double.MaxValue и нулевом Value метод возвращает (0, NaN)

- выражение 2 * Math.PI * freq * Value переписано как 2 * Math.PI * (freq * Value)

Commit:
refactor(Model.Inductor): freq переименован во frequency

Commit:
refactor(Model.IRadioComponent): freq переименован во frequency

Commit:
refactor(Model.Resistor): freq переименован во frequency

Commit:
refactor(Model.Capacitor): freq переименован во frequency

Commit:
fix(Model.UnitTests.InductorTests.GetImpedanceMethodTestCases): исправлена формула вычисления expectedImpedance

- выражение 2 * Math.PI * frequency * radioComponentValue исправлено на 2 * Math.PI * (frequency * radioComponentValue)

Commit:
feature(Model.UnitTests): добавлен тестовый класс CapacitorTests

- реализован метод GetImpedanceMethodTestCases, формирующий тесты метода GetImpedance
- реализован тестовый метод GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance
- добавлены тесты метода GetImpedance: перед методом GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance добавлен атрибут [Test, TestCaseSource("GetImpedanceMethodTestCases")]

Commit:
fix(Model.Capacitor.CalcImpedance): при частоте double.MaxValue и нулевом Value знаменатель дроби -1 / (2 * Math.PI * frequency * Value) становится NaN

- выражение -1 / (2 * Math.PI * frequency * Value) переписано как -1 / (2 * Math.PI * (frequency * Value))

Commit:
feature(Model.Capacitor.CalcImpedance): добавлена проверка Value на равенство нулю

Commit:
style(Model.UnitTests.CapacitorTests): удалены неиспользуемые директивы using

Commit:
feature(Model.Resistor): удален метод double GetImpedance()

iKalentii commented on Testing/Model.UnitTests/CapacitorTests.cs:
```diff
@@ -0,0 +1,73 @@
+
+		[Test, TestCaseSource("GetImpedanceMethodTestCases")]
+		public override void
+			GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance(
```
Не очень удобно будет, если начнёшь все хорошие данные перечислять, если их будет много. Поэтому я бы сократил среднюю часть до GoodParametersAssigned

TemZZZ replied:
Тут я очень много думал, чтобы все комбинации, например, предусмотреть. Тоже обсудить.

iKalentii commented on Testing/Model.UnitTests/IRadioComponentTestsBase.cs:
```diff
@@ -0,0 +1,148 @@
+	/// <typeparam name="T">Класс, реализующий интерфейс
+	/// <see cref="IRadioComponent"/></typeparam>
+	[TestFixture]
+	public abstract class IRadioComponentTestsBase<T>
```
I в названии вводит в заблуждение, конечно. Не думал переделать с наследование на агрегирование, чтобы так сильно не усложнять архитектуру и не вводить необходимость вникания ещё и в тестовую архитектуру. Ещё поясню - вместо наследования, включить этот объект в каждый класс, тестирующий разные элементы.

TemZZZ replied:
Вот этот вопрос нужно обсудить онлайн.

iKalentii commented on Testing/Model.UnitTests/CapacitorTests.cs:
```diff
@@ -0,0 +1,73 @@
+	/// Набор тестов для класса <see cref="Capacitor"/>
+	/// </summary>
+	[TestFixture]
+	class CapacitorTests : IRadioComponentTestsBase<Capacitor>
```
Явно надо писать модификаторы доступа по стандарту RSDN.

iKalentii commented on Testing/Model.UnitTests/CapacitorTests.cs:
```diff
@@ -0,0 +1,73 @@
+using NUnit.Framework;
+using Model.PassiveComponents;
+
+
```
По тестированию - не все публичные части классов покрыты тестами. Если что Coverage проверял тулзой от JetBrains - dotCover.
![image](https://user-images.githubusercontent.com/902806/83282986-a8930e00-a204-11ea-83be-0b7fa02ed3c8.png)

TemZZZ replied:
Тут да, хотел по "чистоте" кода пройтись, все ли учел мой мозг. Без использования катализаторов. Просто пока я не знаю, чем можно пользоваться. Ибо продукты от jetbrains сейчас только ломаные могу скачать. Насколько легально работаем?

iKalentii commented on Model/RadioComponentBase.cs:
```diff
@@ -45,14 +48,27 @@ public double Value
             set
             {
+                if (double.IsNaN(value))
+				{
```
Чёт скобки и отступы поехали

Commit:
refactor(Model.UnitTests.IRadioComponentTestsBase): GetImpedance_GoodFrequencyForRadioComponentWithAssignedGoodValueToValueProperty_ReturnsExpectedImpedance переименован в GetImpedance_GoodParametersAssigned_ReturnsExpectedImpedance

Commit:
refactor(Model.UnitTests.RadioComponentBaseTests): переименован из IRadioComponentTestsBase

Commit:
style(RadioComponentsCollection): исправлено форматирование кода

- в файлах Capacitor.cs, IRadioComponent.cs, RadioComponentBase.cs четырехпробельные отступы в начале строк заменены табуляциями

Commit:
refactor(Model.UnitTests.RadioComponentTests): переименован из RadioComponentBaseTests

Commit:
feature(Model.UnitTests.RadioComponentTests): изменена функциональность класса

- класс теперь не абстрактный
- модификаторы доступа констант MinRadioComponentValue и MinFrequency изменены с protected на public
- модификаторы доступа свойств GoodRadioComponentValues и GoodFrequencies изменены с protected на public
- модификатор доступа метода GetRadioComponent изменен с protected на private
- модификаторы доступа всех статических IEnumerable<TestCaseData> изменены с protected на public
- метод GetImpedance_GoodParametersAssigned_ReturnsExpectedImpedance теперь не виртуальный
- добавлен и реализован метод ConstructorNoParametersTestCase, формирующий тест конструтора без параметров
- добавлен и реализован метод UnitTypeQuantityProperties_Always_ReturnsValues, тестирующий свойства Unit, Type, Quantity
- добавлен и реализован метод Constructor_NoParameters_SetsDefaultRadioComponentValue, тестирующий конструктор без параметров
- удалены все атрибуты Test, TestCaseSource, TestFixture
- изменен комментарий к классу

Commit:
feature(Model.UnitTests.ResistorTests): изменена функциональность класса

- класс теперь ничего не наследует
- добавлено закрытое поле _radioComponentTests - объект класса RadioComponentTests<Resistor> для делегирования ему вызовов тестовых методов и методов-формирователей тестовых случаев классу RadioComponentTests
- добавлены и реализованы публичные методы-формирователи тестовых случаев ValuePropertyGoodValuesTestCases, ValuePropertyBadValuesTestCases, GetImpedanceMethodBadFrequenciesTestCases, ConstructorNoParametersTestCase; эти методы возвращают значения одноименных статических методов класса RadioComponentTests<Resistor>
- добавлены и реализованы публичные тестовые методы ValueProperty_AssignedGoodValue_IsAssigned, ValueProperty_AssignedBadValue_ThrowsExpectedException, GetImpedance_BadFrequency_ThrowsExpectedException, GetImpedance_GoodParametersAssigned_ReturnsExpectedImpedance, UnitTypeQuantityProperties_Always_ReturnsValues, Constructor_NoParameters_SetsDefaultRadioComponentValue; эти методы вызывают одноименные методы класса RadioComponentTests<Resistor> через объект _radioComponentTests
- перед всеми тестовыми методами добавлены соответствующие им атрибуты TestCaseSource и TestCase (добавлены тестовые случаи)

Commit:
feature(Model.UnitTests.InductorTests): изменена функциональность класса

- класс теперь ничего не наследует
- добавлено закрытое поле _radioComponentTests - объект класса RadioComponentTests<Inductor> для делегирования ему вызовов тестовых методов и методов-формирователей тестовых случаев классу RadioComponentTests
- добавлены и реализованы публичные методы-формирователи тестовых случаев ValuePropertyGoodValuesTestCases, ValuePropertyBadValuesTestCases, GetImpedanceMethodBadFrequenciesTestCases, ConstructorNoParametersTestCase; эти методы возвращают значения одноименных статических методов класса RadioComponentTests<Resistor>
- добавлены и реализованы публичные тестовые методы ValueProperty_AssignedGoodValue_IsAssigned, ValueProperty_AssignedBadValue_ThrowsExpectedException, GetImpedance_BadFrequency_ThrowsExpectedException, GetImpedance_GoodParametersAssigned_ReturnsExpectedImpedance, UnitTypeQuantityProperties_Always_ReturnsValues, Constructor_NoParameters_SetsDefaultRadioComponentValue; эти методы вызывают одноименные методы класса RadioComponentTests<Resistor> через объект _radioComponentTests
- перед всеми тестовыми методами добавлены соответствующие им атрибуты TestCaseSource и TestCase (добавлены тестовые случаи)

Commit:
feature(Model.UnitTests.CapacitorTests): изменена функциональность класса

- класс теперь ничего не наследует
- добавлено закрытое поле _radioComponentTests - объект класса RadioComponentTests<Capacitor> для делегирования ему вызовов тестовых методов и методов-формирователей тестовых случаев классу RadioComponentTests
- добавлены и реализованы публичные методы-формирователи тестовых случаев ValuePropertyGoodValuesTestCases, ValuePropertyBadValuesTestCases, GetImpedanceMethodBadFrequenciesTestCases, ConstructorNoParametersTestCase; эти методы возвращают значения одноименных статических методов класса RadioComponentTests<Resistor>
- добавлены и реализованы публичные тестовые методы ValueProperty_AssignedGoodValue_IsAssigned, ValueProperty_AssignedBadValue_ThrowsExpectedException, GetImpedance_BadFrequency_ThrowsExpectedException, GetImpedance_GoodParametersAssigned_ReturnsExpectedImpedance, UnitTypeQuantityProperties_Always_ReturnsValues, Constructor_NoParameters_SetsDefaultRadioComponentValue; эти методы вызывают одноименные методы класса RadioComponentTests<Resistor> через объект _radioComponentTests
- перед всеми тестовыми методами добавлены соответствующие им атрибуты TestCaseSource и TestCase (добавлены тестовые случаи)

Commit:
feature(Model.UnitTests.Constructor_NoParameters_SetsDefaultRadioComponentValue): изменено создание тестируемого объекта

- radioComponent теперь создается конструктором без параметров

Commit:
feature(Model.UnitTests.RadioComponentTests): добавлен и реализован тестовый метод ToString_Always_ReturnsValue

Commit:
test(Model.UnitTests.ResistorTests): добавлен тест метода ToString

Commit:
test(Model.UnitTests.InductorTests): добавлен тест метода ToString

Commit:
test(Model.UnitTests.CapacitorTests): добавлен тест метода ToString

Commit:
refactor(Model.UnitTests.RadioComponentTests): изменен класс

- добавлено ограничение типа new()
- переписан метод GetRadioComponent; новый объект возвращается оператором new()
- переписан метод Constructor_NoParameters_SetsDefaultRadioComponentValue; новый объект создается оператором new T()
- все комментарии // Setup заменены на // Arrange

Commit:
refactor(Model.UnitTests.RadioComponentTests): изменены имена тестовых случаев

- метод ValuePropertyGoodValuesTestCases теперь возвращает TestCaseData, содержащую дополнительно имя класса объекта
- метод ValuePropertyBadValuesTestCases теперь возвращает TestCaseData, содержащую дополнительно имя класса объекта
- метод GetImpedanceMethodBadFrequenciesTestCases теперь возвращает TestCaseData, содержащую дополнительно имя класса объекта
- метод ConstructorNoParametersTestCase теперь возвращает TestCaseData, содержащую дополнительно имя класса объекта
- во всех методах, перечисленных выше, имена свойств, методов и исключений, используемые в формировании имен TestCaseData, переписаны через nameof()
- источники тестовых данных отделены комментарием // TestCaseSources that are common for all radiocomponents

Commit:
refactor(Model.UniotTests.ValueProperty_AssignedGoodValue_IsAssigned): expected изменена на expectedValue, actual изменена на actualValue

Commit:
style(Model.UnitTests): тестовые методы отделены комментарием

Commit:
test(Model.UnitTests.RadioComponentTests): добавлены тесты, общие для всех типов радиокомпонентов

- перед методом ValueProperty_AssignedGoodValue_IsAssigned добавлен источник тестовых данных [TestCaseSource(nameof(ValuePropertyGoodValuesTestCases))]
- перед методом ValueProperty_AssignedBadValue_ThrowsExpectedException добавлен источник тестовых данных [TestCaseSource(nameof(ValuePropertyBadValuesTestCases))]
- перед методом GetImpedance_BadFrequency_ThrowsExpectedException добавлен источник тестовых данных [TestCaseSource(nameof(GetImpedanceMethodBadFrequenciesTestCases))]
- перед методом Constructor_NoParameters_SetsDefaultRadioComponentValue добавлен источник тестовых данных [TestCaseSource(nameof(ConstructorNoParametersTestCase))]

Commit:
test(Model.UnitTests.RadioComponentTests): добавлены типы тестируемых классов

- перед определением класса добавлены атрибуты [TestFixtureSource(typeof(Resistor))],  [TestFixtureSource(typeof(Inductor))],  [TestFixtureSource(typeof(Capacitor))]

Commit:
fix(Model.UnitTests.RadioComponentTests): исправлены неправильные атрибуты перед определением класса

- TestFixtureSource исправлены на TestFixture

Commit:
fix(Model.UnitTests.RadioComponentTests): все nameof(T) заменены на typeof(T).Name

- в именах тестов при использовании nameof(T) имя типа радиокомпонента отображается как T; typeof(T).Name решает проблему, и тип отображается верно

Commit:
fix(Model.UnitTests.RadioComponentTests): все {nameof(doubleToExpectedExceptionType.Value)} заменены на {doubleToExpectedExceptionType.Value.Name}

- неправильно отображались имена исключений

Commit:
fix(Model.UnitTests.UnitTypeQuantityProperties_Always_ReturnsValues): изменены принимаемые параметры

- для корректной работы NUnit лучше не передавать массивы строк как один параметр, он из почему-то пытается развернуть
- изменено тело метода

Commit:
refactor(Model.UnitTests.ResistorTests): изменен класс

- удален ValuePropertyGoodValuesTestCases
- удален ValuePropertyBadValuesTestCases
- удален ConstructorNoParametersTestCase
- удален ValueProperty_AssignedGoodValue_IsAssigned
- удален ValueProperty_AssignedBadValue_ThrowsExpectedException
- удален GetImpedance_BadFrequency_ThrowsExpectedException
- удален Constructor_NoParameters_SetsDefaultRadioComponentValue
- изменена реализация GetImpedanceMethodTestCases: имя теста формируется с использованием nameof соответствующих свойств и методов
- добавлен и реализован метод UnitTypeQuantityPropertiesTestCases
- добавлен и реализован метод ToStringTestCases
- атрибуты TestCase и TestCaseSource переписаны через nameof

Commit:
refactor(Model.UnitTests.InductorTests): изменен класс

- удален ValuePropertyGoodValuesTestCases
- удален ValuePropertyBadValuesTestCases
- удален ConstructorNoParametersTestCase
- удален ValueProperty_AssignedGoodValue_IsAssigned
- удален ValueProperty_AssignedBadValue_ThrowsExpectedException
- удален GetImpedance_BadFrequency_ThrowsExpectedException
- удален Constructor_NoParameters_SetsDefaultRadioComponentValue
- изменена реализация GetImpedanceMethodTestCases: имя теста формируется с использованием nameof соответствующих свойств и методов
- добавлен и реализован метод UnitTypeQuantityPropertiesTestCases
- добавлен и реализован метод ToStringTestCases
- атрибуты TestCase и TestCaseSource переписаны через nameof

Commit:
refactor(Model.UnitTests.CapacitorTests): изменен класс

- удален ValuePropertyGoodValuesTestCases
- удален ValuePropertyBadValuesTestCases
- удален ConstructorNoParametersTestCase
- удален ValueProperty_AssignedGoodValue_IsAssigned
- удален ValueProperty_AssignedBadValue_ThrowsExpectedException
- удален GetImpedance_BadFrequency_ThrowsExpectedException
- удален Constructor_NoParameters_SetsDefaultRadioComponentValue
- изменена реализация GetImpedanceMethodTestCases: имя теста формируется с использованием nameof соответствующих свойств и методов
- добавлен и реализован метод UnitTypeQuantityPropertiesTestCases
- добавлен и реализован метод ToStringTestCases
- атрибуты TestCase и TestCaseSource переписаны через nameof

Commit:
feature(Model.UnitTests.RadiocComponentTests): реализован UnitTypeQuantityPropertiesTestCases

- метод UnitTypeQuantityPropertiesTestCases, использующийся в классах ResistorTests, InductorTests, CapacitorTests вынесен в RadiocComponentTests

Commit:
refactor(Model.UnitTests.RadioComponentTests.UnitTypeQuantityPropertiesTestCases): в имена тестов добавлено имя класса радиокомпонента

fix(Model.UnitTests.RadioComponentTests.ConstructorNoParametersTestCase): исправлена ошибка в именах тестов

Commit:
refactor(Model.UnitTests.ResistorTests.UnitTypeQuantityPropertiesTestCases): вызов метода делегирован классу RadioComponentTests

Commit:
refactor(Model.UnitTests.InductorTests.UnitTypeQuantityPropertiesTestCases): вызов метода делегирован классу RadioComponentTests

Commit:
refactor(Model.UnitTests.CapacitorTests.UnitTypeQuantityPropertiesTestCases): вызов метода делегирован классу RadioComponentTests

Commit:
refactor(Model.UnitTests.CapacitorTests): реализация TestCaseData вынесена из циклов в локальную функцию GetImpedanceTestCaseData

Commit:
style(Model.UnitTests.RadioComponentTests): комментарии заменены регионами

- регионом обозначена облать, озаглавленная комментарием // TestCaseSources that are common for all radiocomponents
- регионом обозначена облать, озаглавленная комментарием // TestMethods that are common for all radiocomponents

Commit:
style(Model.UnitTests.ResistorTests): комментарии заменены регионами

- регионом обозначена облать, озаглавленная комментарием // TestCaseSources
- регионом обозначена облать, озаглавленная комментарием // Tests

Commit:
style(Model.UnitTests.InductorTests): комментарии заменены регионами

- регионом обозначена облать, озаглавленная комментарием // TestCaseSources
- регионом обозначена облать, озаглавленная комментарием // Tests

Commit:
style(Model.UnitTests.CapacitorTests): комментарии заменены регионами

- регионом обозначена облать, озаглавленная комментарием // TestCaseSources
- регионом обозначена облать, озаглавленная комментарием // Tests
